File: HashTable/0242.Valid-Anagram/Solution.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        
        var signatureS = new int[26];
        var signatureT = new int[26];

        foreach (var c in s) signatureS[c-'a']++;
        foreach (var c in t) signatureT[c-'a']++;

        return signatureS.SequenceEqual(signatureT);
    }
}

File: HashTable/0242.Valid-Anagram/Solution.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        
        const int alphabetSize = 26;
        
        if (s.size() != t.size()) return false;
        
        int freqS[alphabetSize] = {0}, freqT[alphabetSize]= {0};
        for (int i = 0; i < s.size(); ++i) {
            freqS[s[i]-'a']++;
            freqT[t[i]-'a']++;
        }
        
        for (int i = 0; i < alphabetSize; ++i) {
            if (freqS[i] != freqT[i]) return false;
        }
        
        return true;
    }
};
